### DISCUSSION TITLE
Ignore touch that wakes the display?

### DISCUSSION THREAD

**Discussion by Tall-Paul** [HASwitchPlate/openHASP#680] [General]
I have a 4" display that's working perfectly now, except for one minor annoyance.  The screen backlight turns off when idle (which is great) but when I touch to wake it I invariably touch one of the buttons on the screen.  Is there a way to tell openhasp to ignore a touch if the backlight is off (other than to turn the backlight on)?

**Comment by dgomes** [comment]
one "trick" is to put a new object on top of all other objects when the device goes to sleep, and remove it when you capture a touch event.

But I agree, we could use a simpler method without any "tricks"

**Comment by fvanroie** [comment]
That's the way that it is supposed to work. Which 4" display are you using?

Please check the serial/telnet logs, there should be a message that reads "First touch enabled/disabled" indicating the first press will be ignored.

**Comment by Tall-Paul** [comment]
I'm using the esp32-s3-4848S040.

I think I see what's happening.  In HA I had the 'backlight when idle' set to 45%.  On this device that just turns off the display completely.  But HASP thinks it's still on.   So it only sets the first touch flag on a long idle.  

If I tell HA to turn off the backlight completely on idle it should presumably set the first touch flag on short idle.

**Reply by fvanroie** [reply]
Yeah, openHASP engages first touch only when the display is technically off.
